src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -30,267 +33,247 @@ const logger = createLoggerWithLabel(module)
-      if (jwtPayloadResult.isErr()) {
-        logger.error({
-          message: 'Failed to verify Corppass JWT with auth client',
-          meta: logMeta,
-          error: jwtPayloadResult.error,
+
+        // NOTE: This should short circuit in the event of an error.
+        // This is why sendSubmissionToAdmin is separated from sendEmailConfirmations in 2 blocks
+        return MailService.sendSubmissionToAdmin({
+          replyToEmails: EmailSubmissionService.extractEmailAnswers(
+            parsedResponses,
+          ),
+          form,
+          submission,
+          attachments,
+          dataCollationData: emailData.dataCollationData,
+          formData: emailData.formData,
         })
-        const { errorMessage, statusCode } = mapRouteError(
-          jwtPayloadResult.error,
-        )
-        return res
-          .status(statusCode)
-          .json({ message: errorMessage, spcpSubmissionFailure: true })
-      }
-      parsedResponses.push(
-        ...createCorppassParsedResponses(
-          jwtPayloadResult.value.userName,
-          jwtPayloadResult.value.userInfo,
-        ),
-      )
-      break
-    }
-    case AuthType.MyInfo: {
-      const uinFinResult = MyInfoUtil.extractMyInfoCookie(req.cookies)
-        .andThen(MyInfoUtil.extractAccessTokenFromCookie)
-        .andThen((accessToken) => MyInfoFactory.extractUinFin(accessToken))
-      if (uinFinResult.isErr()) {
-        const { errorMessage, statusCode } = mapRouteError(uinFinResult.error)
-        return res
-          .status(statusCode)
-          .json({ message: errorMessage, spcpSubmissionFailure: true })
-      }
-      const uinFin = uinFinResult.value
-      const verifyMyInfoResult = await MyInfoFactory.fetchMyInfoHashes(
-        uinFin,
-        formId,
-      ).andThen((hashes) =>
-        MyInfoFactory.checkMyInfoHashes(parsedResponses, hashes),
-      )
-      if (verifyMyInfoResult.isErr()) {
-        logger.error({
-          message: 'Error verifying MyInfo hashes',
-          meta: logMeta,
-          error: verifyMyInfoResult.error,
+          .map(() => ({ form, parsedResponses, submission, emailData }))
+          .mapErr(
+            logErrorWithSubmissionMeta('Error sending submission to admin'),
+          )
+      })
+      // NOTE: The final step returns a ResultAsync so it is not a map
+      .andThen(({ form, parsedResponses, submission, emailData }) => {
+        // MyInfo access token is single-use, so clear it
+        res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
+        // Return the reply early to the submitter
+        res.json({
+          message: 'Form submission successful.',
+          submissionId: submission.id,
         })
-        const { errorMessage, statusCode } = mapRouteError(
-          verifyMyInfoResult.error,
-        )
+
+        // Send email confirmations
+        return SubmissionService.sendEmailConfirmations({
```

Comment:
the intent of the code might be clearer if we prefix this with `void` and shift it above the call to `res.json`, to make it more obvious that we're not awaiting on the email confirmations

Reply:
this would result in a behavioral change from the previous implementation - if an error exists on sending email confirmations, this would not be reflected to the client.

if this is what's desired (which i think it is, because the submission should not affect the webpage view), then i think a better change would be to just segregate this into 2 blocks - `andThen` to send the confirmations and a `map` to send the json response back to the client

Reply:
i combined it into a single `map` block and prefixed it with `void`; main reason is because we need to call `orElse` on `sendEmailConfirmations` which requires typehints. The typehinting here is rather verbose due to the parameter forwarding